### PR TITLE
Feature/mangata liquidity asset as stake

### PR DESCRIPTION
**Pull request by iStrike7**

### CODE REVIEW

**Comment by gleb-urvanov** [comment]
Noticed the big part of the changes is caused by styling, namely by indents. If you can switch to spaces everywhere (which I expect to be enforced by our default format config), it will be easier to see the actual alterations.

File: pallets/staking/src/lib.rs
Comment (gleb-urvanov) [inline pallets/staking/src/lib.rs:2492]: the rewards should be payed in Mantaga tokens, not in liquidity tokens

File: pallets/staking/src/lib.rs
Comment (gleb-urvanov) [inline pallets/staking/src/lib.rs:348]: How and why the dummy value is used?

File: pallets/staking/src/lib.rs
Comment (gleb-urvanov) [inline pallets/staking/src/lib.rs:446]: Why `Staked` is removed in favor of `Stash`?

File: pallets/staking/src/lib.rs
Comment (iStrike7) [reply pallets/staking/src/lib.rs:2492]: Right! Thanks for catching that. will change.

File: pallets/staking/src/lib.rs
Comment (iStrike7) [reply pallets/staking/src/lib.rs:348]: for:
```/// Clears both snapshots of stakers.
    fn kill_stakers_snapshot() {
        <SnapshotValidators<T>>::kill();
        <SnapshotNominators<T>>::kill();

        // Remove Stash's liquidity valuation details
        <StashStakedValuation<T>>::remove_prefix(DUMMY_VALUE);
    }```
    If we don't use DUMMY_VALUE, then we can't use remove_prefix, then we have to remove each item individually.

File: pallets/staking/src/lib.rs
Comment (iStrike7) [reply pallets/staking/src/lib.rs:446]: Since the payout is in MNG, but the staking in liquidity token, we can't at this point support Staked.

File: pallets/staking/src/lib.rs
Comment (iStrike7) [reply pallets/staking/src/lib.rs:2489]: Changed.

File: pallets/staking/src/lib.rs
Comment (gleb-urvanov) [reply pallets/staking/src/lib.rs:348]: Ok, will take a deeper look into it

File: pallets/staking/src/lib.rs
Comment (gleb-urvanov) [inline pallets/staking/src/lib.rs:2530]: `native_liquidity_token` should be renamed into `native_token`, since Mangata token is not a liquidity token

File: pallets/staking/src/lib.rs
Comment (gleb-urvanov) [inline pallets/staking/src/lib.rs:2981]: remove unused code

**Review by gleb-urvanov — CHANGES_REQUESTED** [review]

**Comment by gleb-urvanov** [comment]
also please resolve the conflict

File: node/src/chain_spec.rs
Comment (mateuszaaa) [inline node/src/chain_spec.rs:129]: do we have task for it ?

File: node/src/chain_spec.rs
Comment (mateuszaaa) [inline node/src/chain_spec.rs:317]: can  `(account_id, _, _, _, _, liquidity_token_id, liquidity_token_amount)`  be used directly in map statement ? :thinking: 

```
map(|(account_id, _, _, _, _, liquidity_token_id, liquidity_token_amount)|{
```

File: node/src/chain_spec.rs
Comment (mateuszaaa) [inline node/src/chain_spec.rs:386]: consider moving values extraction into map statement

File: pallets/tokens/src/multi_token_imbalances.rs
Comment (mateuszaaa) [inline pallets/tokens/src/multi_token_imbalances.rs:10]: wouldnt zero be a more appropriate name ?

File: pallets/tokens/src/multi_token_currency.rs
Comment (mateuszaaa) [inline pallets/tokens/src/multi_token_currency.rs:351]: doesnt really seem to be `CurrencyAttribute` isnt there a better to propagate that information?

File: pallets/xyk/src/mock.rs
Comment (mateuszaaa) [inline pallets/xyk/src/mock.rs:72]: now i think first try to mint new currency will fail - as it takes id from the Tokens storage and its default value is `0`. Can we somehow provide proper initial value (`1`) to tokens storage(`NextCurrencyId`) ?

File: runtime/src/lib.rs
Comment (mateuszaaa) [inline runtime/src/lib.rs:287]: it seems like it will evaluate to `0` each time when total_issuance is < u64::max_value() ? is that expected behaviour ?

File: runtime/src/lib.rs
Comment (mateuszaaa) [inline runtime/src/lib.rs:301]: consider using `checked_mul`

File: runtime/src/lib.rs
Comment (mateuszaaa) [inline runtime/src/lib.rs:295]: possible division by zero

**Review by mateuszaaa — COMMENTED** [review]
few minor comments/questions

File: pallets/staking/src/lib.rs
Comment (iStrike7) [reply pallets/staking/src/lib.rs]: done. ty for catching that.

File: pallets/staking/src/lib.rs
Comment (iStrike7) [reply pallets/staking/src/lib.rs]: done.

File: node/src/chain_spec.rs
Comment (iStrike7) [reply node/src/chain_spec.rs:128]: Old no-longer relevant comment. removed.

File: node/src/chain_spec.rs
Comment (iStrike7) [reply node/src/chain_spec.rs:313]: oh yeah, lol... I'll keep it in mind.
Lots of cases in chain_spec, I'll make a mini-task in trello to fix this.

File: node/src/chain_spec.rs
Comment (iStrike7) [reply node/src/chain_spec.rs:382]: Created trello task for it.

File: pallets/tokens/src/multi_token_imbalances.rs
Comment (iStrike7) [reply pallets/tokens/src/multi_token_imbalances.rs:10]: zero already exists as a function in the Imbalance trait

File: runtime/src/lib.rs
Comment (iStrike7) [reply runtime/src/lib.rs:287]: there's .max(1) to prevent that

File: runtime/src/lib.rs
Comment (iStrike7) [reply runtime/src/lib.rs:301]: This is ripped from substrate.

File: runtime/src/lib.rs
Comment (iStrike7) [reply runtime/src/lib.rs:295]: avoided by max(1)

File: pallets/tokens/src/multi_token_currency.rs
Comment (mateuszaaa) [reply pallets/tokens/src/multi_token_currency.rs:351]: after rethinking it really seems to be ok ...

**Review by gleb-urvanov — APPROVED** [review]
Suitable for multivalidator setup. 
However, raises a lot of linter warnings. I would prefer to merge asap and fix warnings later when CD lint/style check will be in place.

Suggest to wait for @mateuszaaa approve as well before merging.

**Review by mateuszaaa — APPROVED** [review]
LGTM :rocket:

**Pull request merged**
